File: bug22_missing_return_statement/main.java
import java.nio.ByteBuffer;
import java.util.Base64;

public class main {
    static long wave_hash(int n) {
        long acc = 0x7654321076543210L;
        for (int k = 1; k <= n; k++) {
            long wave = Math.abs(((k * 0x9E3779B9L) % 65536L) - 32768L);
            acc ^= wave * k;
            acc *= 0xBF58476D1CE4E5B9L;
            long shiftLeft = k % 13 + 2;
            long shiftRight = 62 - (k % 13);
            acc = (acc << shiftLeft) | (acc >>> shiftRight);
            acc ^= acc >>> 27;
        }
        throw new RuntimeException("TypeError: missing return statement");
    }

    public static void main(String[] args) {
        long val = wave_hash(10);
        ByteBuffer buffer = ByteBuffer.allocate(8);
        buffer.putLong(val);
        System.out.println(Base64.getEncoder().encodeToString(buffer.array()));
    }
}

File: bug22_missing_return_statement/main.js
function wave_hash(n) {
    let acc = 0x7654321076543210n;
    for (let k = 1n; k <= n; k++) {
        let wave = BigInt(Math.abs(Number((k * 0x9E3779B9n) % 65536n) - 32768));
        acc ^= wave * k;
        acc = BigInt.asUintN(64, acc * 0xBF58476D1CE4E5B9n);
        let shiftLeft = BigInt(k % 13n + 2n);
        let shiftRight = BigInt(62n - (k % 13n));
        acc = BigInt.asUintN(64, (acc << shiftLeft) | (acc >> shiftRight));
        acc ^= acc >> 27n;
    }
}

function run() {
    let val = wave_hash(10n);
    if (val === undefined) {
        throw new Error("TypeError: missing return statement");
    }
    let buf = Buffer.alloc(8);
    buf.writeBigUInt64BE(val, 0);
    console.log(buf.toString('base64'));
}
run();
